pkg/controller/direct/bigquerydataset/dataset_controller.go

```diff
@@ -303,9 +320,8 @@ func (a *Adapter) Delete(ctx context.Context, deleteOp *directbase.DeleteOperati
 	if err != nil {
 		return false, fmt.Errorf("failed to parse bigquery dataset full name, %w", err)
 	}
-	deleteDatasetCall := a.gcpService.Datasets.Delete(parent.ProjectID, datasetId)
-	err = deleteDatasetCall.Do()
-	if err != nil {
+	dsHandler := a.gcpService.DatasetInProject(parent.ProjectID, datasetId)
+	if err := dsHandler.Delete(ctx); err != nil {
```

Comment:
It seems the old behavior is equivalent to calling `DeleteWithContents`?

Reply:
You are right.